viewing_party/party.py

```diff
@@ -1,23 +1,198 @@
+    user has not watched.
+        Args:
+            user_data (dict): A dict of movies the user has watched and 
+            their friends have watched. 
+        Returns:
+            friend_unique_watched (list): Movies only the users friends 
+            have watched.
+    """
+    friends_unique_watched = []
+
+    for friend in user_data["friends"]:
+        for friend_movie in friend["watched"]:
+            if friend_movie not in user_data["watched"] and (
+            friend_movie not in friends_unique_watched):
+                friends_unique_watched.append(friend_movie)
+    
+    return friends_unique_watched
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    """
+    Return a list of recommended movies for the user based on their subscriptions.
+        Args:
+            user_data (dict): A dict of the users watched movies and subscriptions,
+            and their friends' watched movies.
+        Returns:
+            recommended_movies (list): Movies only the user friends have watched
+            and that are available in the users subscriptions.
+    """
+    recommended_movies = []
+
+    for friend in user_data["friends"]:
+        for friend_movie in friend["watched"]:
+            if friend_movie not in user_data["watched"] and (
+            friend_movie["host"] in user_data["subscriptions"]):
+                    recommended_movies.append(friend_movie)
+
+    return recommended_movies
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    """
+    Return a list of recommended movies for the user based on genre.
+        Args:
+            user_data (dict): A dict of the users watched movies, subscriptions,
+            and favorites, and their friends' watched movies.
+        Returns:
+            recommended_movies (list): Movies of the users most watched genre
+            that the user hasn't watched.
+    """
+    recommended_movies = []
+    user_favorite_genre = get_most_watched_genre(user_data)
+    friends_unique_watched = get_friends_unique_watched(user_data)
+
+    for movie in friends_unique_watched:
+        if movie["genre"] == user_favorite_genre:
+            recommended_movies.append(movie)
+
+    return recommended_movies
+
+
+def get_rec_from_favorites(user_data):
+    """
+    Return a list of recommended movies based on the users favorites.
+        Args:
+            user_data (dict): A dict of the users watched movies, subscriptions,
+            and favorites, and their friends' watched movies.
+        Returns:
+            recommended_movies (list): Movies in the users favorites that their
+            friends haven't watched.
+    """
+    recommended_movies = []
+    
+    user_unique_watched = get_unique_watched(user_data)

```

Comment:
👍 Nice reuse of our function to get the list of movies that only the user has watched to simplify the logic here.